packages/wonder-blocks-dropdown/components/select-box.js

```diff
@@ -0,0 +1,231 @@
+                                        ? stateStyles.active
+                                        : (hovered || focused) &&
+                                          stateStyles.focus),
+                                style,
+                            ]}
+                            {...handlers}
+                        >
+                            <LabelMedium style={[textColor]}>
+                                {children}
+                            </LabelMedium>
+                            <View style={[styles.spacing]} />
+                            <View style={[styles.caretWrapper]}>
+                                {this.renderCaret()}
+                            </View>
+                        </StyledButton>
+                    );
+                }}
+            </ClickableBehavior>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    shared: {
+        position: "relative",
+        display: "inline-flex",
+        alignItems: "center",
+        justifyContent: "space-between",
+        height: 40,
+        marginTop: 4,
+        marginBottom: 4,
+        paddingLeft: 16,
+        paddingRight: 12,
+        border: "none",
+        borderRadius: 4,
+        cursor: "pointer",
+        outline: "none",
+        textDecoration: "none",
+        boxSizing: "border-box",
+        whiteSpace: "nowrap",
+    },
+
+    textDisabled: {
+        color: offBlack32,
+    },
+
+    placeholder: {
+        color: offBlack50,
+    },
+
+    placeholderDisabled: {
+        color: offBlack16,
+    },
+
+    // consider using Strut
+    spacing: {
+        width: 8,
+    },
+});
+
+const _generateStyles = (light, hovered, focused, pressed) => {
+    let newStyles = {};
+    if (light) {
+        newStyles = {
+            focus: {
+                ":before": {
+                    content: "''",
+                    position: "absolute",
+                    top: -3,
+                    left: -3,
+                    right: -3,
+                    bottom: -3,
+                    borderColor: white,
+                    borderStyle: "solid",
+                    borderWidth: 2,
+                    borderRadius: 7,
+                },
+            },
+            active: {
+                background: mix(fade(blue, 0.32), white),
+            },
+            disabled: {
+                cursor: "auto",
+            },
+        };
+    } else {
+        newStyles = {
+            default: {
+                borderColor: offBlack50,
+                borderStyle: "solid",
+                borderWidth: 1,
+            },
+            focus: {
+                borderColor: blue,
+                borderStyle: "solid",
+                borderWidth: 2,
+                // These values are default padding (16 and 12) minus 1, because
+                // changing the borderWidth to 2 messes up the button width
+                // and causes it to move a couple pixels. This fixes that.
```

Comment:
Thanks for adding a comment.  I'm confused as to why we minus 1 instead of `2` which is the actual border width.  I see that `Button` does the same thing.

Reply:
I think it's because the default state has a border of width 1. In the focused state, the border width changes to 2, so the difference is 1, and that's the amount we need to adjust the padding.

Reply:
That makes sense.  I forgot that we make the focus ring thicker in that case.